src/main/java/org/opensearch/neuralsearch/processor/AgenticQueryTranslatorProcessor.java

```diff
@@ -188,6 +191,66 @@ private void executeAgentAsync(
         }));
     }
 
+    private void preserveOriginalParameters(SearchSourceBuilder original, SearchSourceBuilder target) {
```

Comment:
1. Can you not use `shallowCopy` function?
2. Do you need all those condition checks? Some fields for example with null by default, then doing null check and assigning null directly makes no difference, right? Asking because this function introduces many branches and UT doesn't seem to cover all branches, and to cover them all requires quite some efforts.